src/presenced.cpp

```diff
@@ -342,21 +355,36 @@ Client::reconnect(const std::string& url)
                                                     (struct sockaddr *)&ipv4addr, sizeof(ipv4addr),
                                                     mUrl.port, (mUrl.path).c_str())), "connect");
                 }
+                return promise::_Void();
             })
-            .fail([this](const promise::Error& err)
+            .fail([wptr, this](const promise::Error& err)
             {
-                int errType =  (err.type() == ERRTYPE_MEGASDK) ? WS_ERRTYPE_DNS : err.type();
-                mConnectPromise.reject(err.msg(), err.code(), errType);
-                mLoginPromise.reject(err.msg(), err.code(), errType);
+                PRESENCED_LOG_ERROR("Connect error: %s", err.what());
+                if (wptr.deleted())
+                    return;
```

Comment:
Nop, in both places. You need to handle the error eventually, i.e. not return it further. If you don't intend to attach any further error handlers (like in this case), then you should not propagate the error further, or you will get the "unhanded promise fail" warning.